nixos/modules/services/web-servers/caddy.nix

```diff
@@ -50,28 +79,37 @@ in {
         The data directory, for storing certificates. Before 17.09, this
         would create a .caddy directory. With 17.09 the contents of the
         .caddy directory are in the specified data directory instead.
+
+        Caddy v2 replaced CADDYPATH with XDG directories.
+        See https://caddyserver.com/docs/conventions#file-locations.
       '';
     };
 
     package = mkOption {
-      default = pkgs.caddy;
-      defaultText = "pkgs.caddy";
+      default = pkgs.caddy2;
```

Comment:
Hm, I'm wondering about this. Changing old configurations to `caddy2` is a breaking change, but at the same time, I'm well aware that Caddy probably won't be supported during 20.09 release lifetime, and probably will be removed.

I think defaulting to Caddy 2 is the right choice, but I would appreciate having the validation with `caddy validate` to inform about breakage during 20.03 to 20.09 upgrade.

Reply:
@xfix Thanks, that's very helpful!

Could you please give me some guidance on how to plug `caddy validate` into all this, and properly report validation errors to the user? How can I check if, for instance, `runCommand` call was successful?

Reply:
To be fair, I don't exactly know how to use Nix, but you may want to take a look at `pkgs/build-support/writers/default.nix`.

I believe that any command that returns an error code will interrupt the build. `caddy validate` returns error code 1 on an error, so it's sufficient to run `caddy validate`.

Reply:
@xfix So just failing the build is good enough? Okay, thanks!